### PR TITLE
Add rate limiting for public env

### LINKED ISSUE #38: Container registry rate-limit protection
As a container registry administrator, I want to be able to set rate-limit protection to prevent DoS attacks (see [docker-hub rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) as an example).

Note that this feature is not available directly in the Harbor project, see open [issue](https://github.com/goharbor/harbor/issues/3419).

As the k8s-harbor project is intended to deploy Harbor into k8s cluster, the k8s ingress controller is probably the right place to achieve this kind of protection on Harbor.

### PR DESCRIPTION
**Pull request by chess-knight**
Limit is applied in front of Harbor - on the ingress-nginx

Fixes #38

As we can see [DockerHub overall limit](https://docs.docker.com/docker-hub/download-rate-limit/#other-limits) and [Quay.io rate limit](https://access.redhat.com/articles/5531191) there is thousands of requests per minute respectively tens of requests per second. For the Project Quay, nginx limits are mentioned [here](https://github.com/quay/quay/blob/master/conf/nginx/rate-limiting.conf.jnj). Its about 50r/s based on [location](https://github.com/quay/quay/blob/master/conf/nginx/server-base.conf.jnj).

This harbor instance is a little less used, so 25 limit per second seems to be fine here. This limit is also only a local limit, in the case of n ingress controllers pods, limit = limit * n. In the edge case, we are also fine with docker theoretical max layer depth limit, which is [125](https://github.com/moby/moby/blob/master/layer/layer_store.go#L28) because burts rate limit is set to limit-rps * 5(multiplier can be changed). So in this burst case, we are able to process 1 + 25 * 5 requests within 40ms.

### CODE REVIEW

File: envs/public/harbor-config.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ data:
         harbor:
           annotations:
             cert-manager.io/issuer: letsencrypt
+            nginx.ingress.kubernetes.io/limit-rps: "25"
```
Comment (matofeder) [inline envs/public/harbor-config.yaml:36]: Nginx doc says that NGINX returns status code 503 Service Unavailable when the limit is reached. 
If it is a default in the case above too, consider setting 429 instead.

Comment (matofeder) [inline envs/public/harbor-config.yaml:36]: Document this change in the `docs` of this project. 
- Mention, that Harbor itself does not support this feature and should be handled "outside" e.g. in the ingress controller in this case 
- An example with a real-world scenario could also help to understand how the burst default setting (5) works.

Comment (chess-knight) [reply envs/public/harbor-config.yaml:36]: This status code can be changed via ingress nginx config map, I added an example in the docs. See eb378bf06c29cf4e91f5e123d12592598d5bd278

Comment (chess-knight) [reply envs/public/harbor-config.yaml:36]: See eb378bf06c29cf4e91f5e123d12592598d5bd278, where examples and valuable links show how the nginx rate-limiting works.

**Review by matofeder — APPROVED** [review]
LGTM! 
Thank you that you provide docs page with useful links!

**Pull request merged**